Runtime/Tracking/Modification/TransformPropertyApplier.cs

```diff
@@ -152,7 +152,11 @@ public class UnityEvent : UnityEvent<EventData>
         /// <param name="source">The data to build the new source from.</param>
         public virtual void SetSource(GameObject source)
         {
-            Source = source != null ? new TransformData(source) : null;
+            sourceTransformData.Transform = source != null ? source.transform : null;

```

Comment:
I'm not sure if this is the same logic.

`Source` is an injected object that then builds a representation of overrides within sourceTransformData (i think)

This is actually overriding the injected Source object with the internal representation of that object.

But then I'm struggling to see where `sourceTransformData` is even used. I'll need to investigate more